File: app/src/main/java/org/wikipedia/analytics/IntentFunnel.java
package org.wikipedia.analytics;

import org.json.JSONObject;
import org.wikipedia.WikipediaApp;

import androidx.annotation.NonNull;

public class IntentFunnel extends Funnel {
    private static final String SCHEMA_NAME = "MobileWikiAppIntents";
    private static final int REV_ID = 18115555;

    public IntentFunnel(WikipediaApp app) {
        super(app, SCHEMA_NAME, REV_ID);
    }

    public void logSearchWidgetTap() {
        log(
                "action", "searchwidgettap"
        );
    }

    public void logFeaturedArticleWidgetTap() {
        log(
                "action", "featuredarticlewidgettap"
        );
    }

    public void logShareIntent() {
        log(
                "action", "share"
        );
    }

    public void logProcessTextIntent() {
        log(
                "action", "processtext"
        );
    }

    @Override protected void preprocessSessionToken(@NonNull JSONObject eventData) { }
}

File: app/src/main/java/org/wikipedia/analytics/IntentFunnel.kt
package org.wikipedia.analytics

import org.json.JSONObject
import org.wikipedia.WikipediaApp

class IntentFunnel(app: WikipediaApp) : Funnel(app, SCHEMA_NAME, REV_ID) {

    fun logSearchWidgetTap() {
        log("action" to "searchwidgettap")
    }

    fun logFeaturedArticleWidgetTap() {
        log("action" to "featuredarticlewidgettap")
    }

    fun logShareIntent() {
        log("action" to "share")
    }

    fun logProcessTextIntent() {
        log("action" to "processtext")
    }

    override fun preprocessSessionToken(eventData: JSONObject) {}

    companion object {
        private const val SCHEMA_NAME = "MobileWikiAppIntents"
        private const val REV_ID = 18115555
    }
}
